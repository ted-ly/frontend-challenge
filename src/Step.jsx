import React from 'react';
import Spinner from './Spinner';

export default function Step({ title, children, isLoading }) {
    return (
        <div className='step-container'>
            <div className='step-title'>{title}</div>
            {isLoading ? <Spinner /> : children}
        </div>
    );
}