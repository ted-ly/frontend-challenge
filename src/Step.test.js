import renderer from 'react-test-renderer';
import Step from './Step';

describe('Step', () => {
    it('should match snapshot', () => {
        const component = renderer.create(<Step />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with Spinner', () => {
        const component = renderer.create(<Step isLoading={true}><div>test</div></Step>);
        expect(component.toJSON()).toMatchSnapshot();
    });
});