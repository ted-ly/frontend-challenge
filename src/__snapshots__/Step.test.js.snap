// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step should match snapshot 1`] = `
<div
  className="step-container"
>
  <div
    className="step-title"
  />
</div>
`;

exports[`Step should match snapshot with Spinner 1`] = `
<div
  className="step-container"
>
  <div
    className="step-title"
  />
  <div
    className="Spinner"
  >
    <div
      className="Spinner__bounce1"
    />
    <div
      className="Spinner__bounce2"
    />
    <div
      className="Spinner__bounce3"
    />
  </div>
</div>
`;
